playbooks/roles/buildbot-master/files/master.cfg

```diff
@@ -0,0 +1,201 @@
+                            builderNames=["runtests"]))
+c['schedulers'].append(schedulers.ForceScheduler(
+                            name="force_configuration_tests",
+                            builderNames=["run_configuration_tests"]))
+
+####### BUILDERS
+
+# The 'builders' list defines the Builders, which tell Buildbot how to perform a build:
+# what steps, and which workers can execute them.  Note that any particular build will
+# only take place on one worker.
+
+factory = util.BuildFactory()
+# check out the source
+factory.addStep(steps.Git(repourl='git://github.com/buildbot/pyflakes.git', mode='incremental'))
+# run the tests (note that this will require that 'trial' is installed)
+factory.addStep(steps.ShellCommand(command=["trial", "pyflakes"]))
+
+# Factory for testing configuration repo
+config_test_factory = util.BuildFactory()
+config_test_factory.addStep(steps.Git(
+    name='clone git repo',
+    repourl='git://github.com/Stanford-Online/configuration.git',
+    mode='incremental',
+))
+config_test_factory.addStep(steps.ShellSequence(
+    name='install dependencies via apt-get',
+    commands=[
+        util.ShellArg(command=['sudo', 'apt-get', 'update']),
+        util.ShellArg(command=['sudo', 'apt-get', 'install', '-y', 'nodejs']),
+        util.ShellArg(command=['sudo', 'apt-get', 'install', '-y', 'python-demjson']),
+    ],
+))
+config_test_factory.addStep(steps.ShellSequence(
+    name='install dependencies via pip',
+    commands=[
+        util.ShellArg(command=['sudo', 'pip', 'install', '--allow-all-external', '-r', 'requirements.txt']),
+        util.ShellArg(command=['sudo', 'pip', 'install', '--allow-all-external', 'demjson']),
+    ],
+))
+config_test_factory.addStep(steps.ShellCommand(
+    name='test YAML files',
+    command=(
+        'for yml in $(find . -name "*.yml"); do\n'
+          'python -c "import sys,yaml; yaml.load(open(sys.argv[1]))" $yml >/dev/null\n'
+          'if [ $? -ne 0 ]; then\n'
+            'echo "ERROR parsing $yml"\n'
+            'exit 1\n'
+          'fi\n'
+        'done'
+    ),
+))
+config_test_factory.addStep(steps.ShellCommand(
+    name='lint JSON files with jsonlint',
+    command=(
+        'for json in $(find . -name "*.json"); do\n'
+          'jsonlint -v $json\n'
+          'if [ $? -ne 0 ]; then\n'
+            'echo "ERROR parsing $json"\n'
+            'exit 1\n'
+          'fi\n'
+        'done'
+    ),
+))
+config_test_factory.addStep(steps.ShellCommand(
+    name='jinja tests',
+    workdir='build/playbooks',
+    command=(
+        'for j2 in $(git diff --name-only refs/heads/master..$(git rev-parse --abbrev-ref HEAD) |grep -E \'.+\\.j2$|.+\\.yml\'); do\n'
+          'python ../tests/jinja_check.py ../$j2\n'
+          'if [ $? -ne 0 ]; then\n'
+            'echo "Jinja parsing error on $j2"\n'
+            'exit 1\n'
+          'fi\n'
+        'done'
+    ),
+))
+config_test_factory.addStep(steps.ShellSequence(
+    name='core tests',
```

Comment:
I'm not really sure about naming conventions for Buildbot build steps. The one example I found uses a short name, all lowercase, with words separated by underscores. I believe there is also a `description` field that says what text to display at the start of the step and a `descriptionDone` field that says what text to display at the end of the step. It doesn't look like most people use those, but I don't really know how much detail we want.